src/object/clone.ts

```diff
@@ -26,48 +26,103 @@
  * console.log(clonedObj); // { a: 1, b: 'es-toolkit', c: [1, 2, 3] }
  * console.log(clonedObj === obj); // false
  */
-export function clone<T>(obj: T): T {
+export function clone<T>(obj: T): Shallowed<T> {
```

Comment:
I think this type looks not as readable to me. What about making use of TypeScript's function overloading, like:

```tsx
export function clone<T>(obj: T[]): T[];
export function clone(obj: Date): Date;
export function clone(obj: RegExp): RegExp;

/* ... overloading for other native objects ... */

export function clone<T>(obj: T): Shallowed<T>;
```

```tsx
type Shallowed<T> = OmitNever<{
           [P in keyof T]: T[P] extends Function ? never : T[P];
         }>
```

Reply:
I think `Shallowed<T>` type is better than multiple function overloading definitions.